soda/core/soda/execution/check/profile_columns_run.py

```diff
@@ -46,96 +47,88 @@ def __init__(self, data_source_scan: DataSourceScan, profile_columns_cfg: Profil
+                f"data source. Please make sure that the patterns in your profiling expressions define existing dataset name + column name combinations."
+                f" Profiling results may be incomplete or entirely skipped. See the docs for more information: \n"
+                f"https://docs.soda.io/soda-cl/profile.html#display-profile-information-in-soda-cloud",
                 location=self.profile_columns_cfg.location,
             )
             return profile_columns_result
-        parsed_included_tables_and_columns = self._build_column_expression_list(
-            self.profile_columns_cfg.include_columns
-        )
-        parsed_excluded_tables_and_columns = self._build_column_expression_list(
-            self.profile_columns_cfg.exclude_columns
-        )
 
         self.logs.info("Profiling columns for the following tables:")
-        for table_name in table_names:
+        for table_name in tables_columns_metadata:
             self.logs.info(f"  - {table_name}")
             profile_columns_result_table = profile_columns_result.create_table(
                 table_name, self.data_source.data_source_name, row_count=None
             )
-            included_columns, excluded_columns = self.build_column_inclusion_exclusion_lists(
-                table_name, parsed_included_tables_and_columns, parsed_excluded_tables_and_columns
-            )
-            # get columns & metadata for current table
-            columns_metadata_result = self.data_source.get_table_columns(
-                table_name=table_name,
-                query_name=f"profile-columns-get-column-metadata-for-{table_name}",
-                included_columns=included_columns,
-                excluded_columns=excluded_columns,
-            )
-            if columns_metadata_result is not None:
-                # perform numerical metrics collection
-                numerical_columns = {
-                    col_name: data_type
-                    for col_name, data_type in columns_metadata_result.items()
-                    if data_type in self.data_source.NUMERIC_TYPES_FOR_PROFILING
-                }
-
-                for column_name, column_type in numerical_columns.items():
-                    try:
-                        self.profile_numeric_column(
-                            column_name,
-                            column_type,
-                            table_name,
-                            profile_columns_result_table,
-                        )
-                    except Exception as e:
-                        self.logs.error(
-                            f"Problem profiling numeric column {table_name}.{column_name}: {e}",
-                            exception=e,
-                        )
+            columns_metadata_result = tables_columns_metadata.get(table_name)
+            # perform numerical metrics collection
+            numerical_columns = {
```

Comment:
while changing this chunk, what about creating 2 methods called `handle_numerical_columns` and `handle_text_columns` to clean this big method.